azurerm/internal/services/media/media_streaming_policy_resource.go

```diff
@@ -0,0 +1,693 @@
+		parameters.NoEncryption = expandNoEncryption(noEncryption.([]interface{}))
+	}
+
+	if commonEncryptionCENC, ok := d.GetOk("common_encryption_cenc"); ok {
+		parameters.CommonEncryptionCenc = expandCommonEncryptionCenc(commonEncryptionCENC.([]interface{}))
+	}
+
+	if commonEncryptionCBCS, ok := d.GetOk("common_encryption_cbcs"); ok {
+		parameters.CommonEncryptionCbcs = expandCommonEncryptionCbcs(commonEncryptionCBCS.([]interface{}))
+	}
+
+	if contentKeyPolicyName, ok := d.GetOk("default_content_key_policy_name"); ok {
+		parameters.DefaultContentKeyPolicyName = utils.String(contentKeyPolicyName.(string))
+	}
+
+	if _, err := client.Create(ctx, resourceID.ResourceGroup, resourceID.MediaserviceName, resourceID.Name, parameters); err != nil {
+		return fmt.Errorf("Error creating Streaming Policy %q in Media Services Account %q (Resource Group %q): %+v", resourceID.Name, resourceID.MediaserviceName, resourceID.ResourceGroup, err)
+	}
+
+	d.SetId(resourceID.ID())
+
+	return resourceMediaStreamingPolicyRead(d, meta)
+}
+
+func resourceMediaStreamingPolicyRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Media.StreamingPoliciesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.StreamingPolicyID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.MediaserviceName, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Streaming Policy %q was not found in Media Services Account %q and Resource Group %q - removing from state", id.Name, id.MediaserviceName, id.ResourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error retrieving Streaming Policy %q in Media Services Account %q (Resource Group %q): %+v", id.Name, id.MediaserviceName, id.ResourceGroup, err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("media_services_account_name", id.MediaserviceName)
+
+	if props := resp.StreamingPolicyProperties; props != nil {
+		noEncryption := flattenNoEncryption(resp.NoEncryption)
+		if err := d.Set("no_encryption_enabled_protocols", noEncryption); err != nil {
+			return fmt.Errorf("Error flattening `no_encryption_enabled_protocols`: %s", err)
+		}
+
+		commonEncryptionCENC := flattenCommonEncryptionCenc(resp.CommonEncryptionCenc)
+		if err := d.Set("common_encryption_cenc", commonEncryptionCENC); err != nil {
+			return fmt.Errorf("Error flattening `common_encryption_cenc`: %s", err)
+		}
+
+		commonEncryptionCBCS := flattenCommonEncryptionCbcs(resp.CommonEncryptionCbcs)
+		if err := d.Set("common_encryption_cbcs", commonEncryptionCBCS); err != nil {
+			return fmt.Errorf("Error flattening `common_encryption_cbcs`: %s", err)
+		}
+
+		if resp.DefaultContentKeyPolicyName != nil {
+			d.Set("default_content_key_policy_name", *resp.DefaultContentKeyPolicyName)
+		}
```

Comment:
in this instance we can remove this `if` statement (and the de-reference) here, `d.Set` will take care of assigning an empty string should this be nil

```suggestion
		d.Set("default_content_key_policy_name", props.DefaultContentKeyPolicyName)
```

(however we should also use `props` and not `resp` since this is nested within the `properties` block)